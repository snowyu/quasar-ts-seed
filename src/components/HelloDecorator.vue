<!-- src/components/HelloDecorator.vue -->
<!-- This is an alternative way to define the Hello component using decorators -->
<template>
  <q-page class="flex flex-center">
    <q-list no-border>
      <q-item class="greeting">Hello Decorator {{name}}{{exclamationMarks}}</q-item>
      <q-btn @click="decrement">-</q-btn>
      <q-btn @click="increment">+</q-btn>
    </q-list>
  </q-page>
</template>

<script lang="ts">
import { Vue, Component, Prop } from "vue-property-decorator";

@Component
export default class HelloDecorator extends Vue {
    @Prop() name!: string;
    @Prop({default: 1}) initialEnthusiasm!: number;
    enthusiasm = this.initialEnthusiasm;
    increment() {
        this.enthusiasm++;
    }
    decrement() {
        if (this.enthusiasm > 1) {
            this.enthusiasm--;
        }
    }
    get exclamationMarks(): string {
        return Array(this.enthusiasm + 1).join('!');
    }
}
</script>

<style>
.greeting {
    font-size: 20px;
}
</style>
