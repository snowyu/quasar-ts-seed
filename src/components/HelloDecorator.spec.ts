/* eslint-disable */
/**
 * @jest-environment jsdom
 */

import { shallowMount, mount, createLocalVue } from '@vue/test-utils';
// import * as Quasar from 'quasar';
import HelloDecorator from './HelloDecorator.vue';

describe('HelloDecorator.vue', () => {
  it('renders HelloDecorator test', () => {
    // const localVue = createLocalVue();
    // localVue.use(Quasar, { components: ['QBtn']});
    const l = 1;
    const wrapper = shallowMount(HelloDecorator, {
      propsData: { title: 'test' }
    });
    const vGreeting = wrapper.find('.greeting');
    expect(vGreeting.text()).toEqual('Hello Decorator test!');
  })

  it('click to increment and decrement', () => {
    const l = 1;
    const wrapper = mount(HelloDecorator, {
      propsData: { title: 'test' }
    });
    const vBtnInc = wrapper.find('#btnInc');
    const vBtnDec = wrapper.find('#btnDec');
    vBtnInc.trigger('click');
    const vGreeting = wrapper.find('.greeting');
    expect(vGreeting.text()).toEqual('Hello Decorator test!!');
    vBtnInc.trigger('click');
    expect(vGreeting.text()).toEqual('Hello Decorator test!!!');
    vBtnDec.trigger('click');
    expect(vGreeting.text()).toEqual('Hello Decorator test!!');
    vBtnDec.trigger('click');
    expect(vGreeting.text()).toEqual('Hello Decorator test!');
    vBtnDec.trigger('click');
    expect(vGreeting.text()).toEqual('Hello Decorator test!');
  })

  it('matches snapshot', () => {
    const name = 'snap';
    let wrapper = shallowMount(HelloDecorator, {
      propsData: { title: name }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
