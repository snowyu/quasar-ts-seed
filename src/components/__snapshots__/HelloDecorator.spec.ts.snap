// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloDecorator.vue matches snapshot 1`] = `
<qlist-stub noborder="true">
  <qitem-stub tag="div" class="greeting">Hello Decorator snap!</qitem-stub>
  <qbtn-stub align="center" id="btnDec">-</qbtn-stub>
  <qbtn-stub align="center" id="btnInc">+</qbtn-stub>
</qlist-stub>
`;
